### PR TITLE
Attempt to batch toggleGroupDataItems with redux-batched-actions

### PR DESCRIPTION
**Pull request by klappy**
#### Describe what your pull request addresses (Please include relevant issue numbers):
https://github.com/unfoldingWord-dev/translationCore/issues/4842
Addresses slow loading of the book of Acts in the Alignment tool.

#### Please include detailed Test instructions for your pull request:
- Load the book of Acts from this project: https://git.door43.org/lrsallee/en_act.git
- Load the wordAlignment tool and ensure it loads both under 20s and menu acts as expected.
- Other projects, tools, loading might be addressable in a similar way.
- Load other projects and tools and ensure all works well.

#### Describe what your pull request addresses (Please include relevant issue numbers):
- Performance issues with drag and drop in large books in wA tool. https://github.com/unfoldingWord-dev/translationCore/issues/4842
- Joel's refactor of wA already triggers the menu update and the workaround that was there is no longer necessary.
- Long term: The whole caching of the menu and refresh counter needs to be removed as it would be nearly as performant to just batch the actions to populate the reducer that feeds the menu. Cache invalidation is one of the hardest things and it is recommended to remove the caching and optimize the code.
https://twitter.com/codinghorror/status/506010907021828096?lang=en

#### Please include detailed Test instructions for your pull request:
- Load a large book with alignment such as Acts from: https://git.door43.org/lrsallee/en_act.git
- Drag and drop words and it should respond in less than 2 seconds instead of 20-30s.

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/4874)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (736b45c)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-736b45c.zip) is ready for download.
[macOS build 1.0.0 (736b45c)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-736b45c.dmg) is ready for download.
[Windows x64 build 1.0.0 (736b45c)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-736b45c.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (736b45c)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-736b45c.setup.exe) is ready for download.

**Review by mannycolon — APPROVED** [review]
:lgtm:

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/4874#-:-LJRTb0JBap6ob0yGosS:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @PhotoNomad0 and @neutrinog)

File: src/js/actions/GroupsDataActions.js [hunk 1]
```diff
@@ -223,44 +227,46 @@ function contains(object, arrayToBeChecked) {
...
         contextId: fileObject.contextId,
         boolean: fileObject.invalidated
-      });
+      };
       break;
     default:
       console.warn("Undefined label in toggleGroupDataItems switch");
       break;
   }
+  return action;
```
Comment (da1nerd) [inline src/js/actions/GroupsDataActions.js]: Action will be `undefined` if the default case is matched.
You should provide a default value for action.

**Review by da1nerd — CHANGES_REQUESTED** [review]

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (1656dea)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-1656dea.zip) is ready for download.
[macOS build 1.0.0 (1656dea)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-1656dea.dmg) is ready for download.
[Windows x64 build 1.0.0 (1656dea)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-1656dea.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (1656dea)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-1656dea.setup.exe) is ready for download.

**Review by da1nerd — APPROVED** [review]

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (8e76ccd)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-8e76ccd.zip) is ready for download.
[macOS build 1.0.0 (8e76ccd)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-8e76ccd.dmg) is ready for download.
[Windows x64 build 1.0.0 (8e76ccd)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-8e76ccd.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (8e76ccd)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-8e76ccd.setup.exe) is ready for download.

**Comment by da1nerd** [comment]
This also resolves https://github.com/unfoldingWord-dev/translationCore/issues/4673

**Review by da1nerd — COMMENTED** [review]
> Reviewed 1 of 1 files at r2, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/4874)* status: all files reviewed, 1 unresolved discussion (waiting on @neutrinog and @PhotoNomad0)

**Comment by da1nerd** [comment]
Confirmed this works well on Linux.

**Comment by dsm-git** [comment]
[Linux build 1.0.0 (ca376d9)](https://tc-builds.door43.org/translationCore-linux-x64-1.0.0-ca376d9.zip) is ready for download.
[macOS build 1.0.0 (ca376d9)](https://tc-builds.door43.org/translationCore-macos-x64-1.0.0-ca376d9.dmg) is ready for download.
[Windows x64 build 1.0.0 (ca376d9)](https://tc-builds.door43.org/translationCore-win-x64-1.0.0-ca376d9.setup.exe) is ready for download.
[Windows x32 build 1.0.0 (ca376d9)](https://tc-builds.door43.org/translationCore-win-x32-1.0.0-ca376d9.setup.exe) is ready for download.

**Comment by PhotoNomad0** [comment]
CLosing - Needs to be based off of release 0.10.0

**Comment by PhotoNomad0** [comment]
Replaced with https://github.com/unfoldingword-dev/translationcore/pull/4882

**Comment by da1nerd** [comment]
@klappy do you need this branch anymore? If not could you delete it?

**Pull request closed without merging**